### ISSUE TITLE
hmmsearch not parallelized

### ISSUE THREAD

**Issue by jianshu93** [shafferm/DRAM#66]
dear DRAM team,

I found that the hmmsearch is really slow and ran just in 2 threads but I have 200 genomes to search. Would that be possible to make it run in multiple processes for each genome at the same time? This is what other softwares is doing now.

Thanks,

Best,

Jianshu

**Comment by houjialin** [comment]
Yes, I also meet this problem, which makes the thresholds are not utilized efficiently. It will spend about a half-hour to annotate one MAG in my sever even I defined 60 thresholds to run DRAM.

**Comment by shafferm** [comment]
Yes hmmsearch most of the time only uses 2 threads even when you give it 60. I am working switching to hh-suite which should get around this problem. Annotating multiple genomes at once would require a total change in architecture, using less threads than mmseqs2 can take advantage of and cause a few other problems as well.

If you have lots of memory available then there are others running multiple instances of `DRAM.py annotate` concurrently. This means you would need to be able to hold in memory the number of process x the amount of memory used by your largest database at once. Alternately if you want to skip annotating with VOGdb to speed things up you can remove it from your CONFIG file by setting `"vogdb": null`.

**Comment by rmFlynn** [comment]
Admittedly, I have a different take on this issue. There is more and more interest in using multiple databases and there is not one size fits all search tool. With distributed computing, threads in a monolithic process are not desirable, we want actual processes that can be distributed. Finally, its not just searches that would benefit from running in parallel, many processes would.
